front-end/src/components/Pricing/Pricing.jsx

```diff
@@ -38,20 +53,34 @@ const Pricing = () => {
 
   useEffect(() => {
     getPricing();
-  }, []);
-
-  const pricingCards = isAuthenticated && currentUser.role === 'student' && (
-    <>
-      <TeacherSelect user={currentUser} chooseTeacher={handleTeacher} teacher={teacher} />
-      <div className={styles.pricing}>
-        {pricing.map((item) => (
-          <PricingCard key={item._id} pricing={item} user={userData} teacher={teacher} />
-        ))}
-      </div>
-    </>
-  );
+    fetchSubscriptions(currentUser.roleId);
+  }, [currentUser]);
+
+  const studentRole = isAuthenticated && currentUser.role === 'student';
```

Comment:
You can get current user only when you authentificated already.

Reply:
fixed